### PR TITLE
Implement fee collection logic

### PR DESCRIPTION
**Pull request by jelysn**

### CODE REVIEW

File: x/amm/keeper/fee.go [hunk 1]
```diff
@@ -16,9 +19,44 @@ func PortionCoins(coins sdk.Coins, portion sdk.Dec) sdk.Coins {
 	return portionCoins
 }
 
-func (k Keeper) OnCollectFee(ctx sdk.Context, pool types.Pool, fee sdk.Coins) {
-	// TODO: transfer 60% to LPs through commitment module
-	// TODO: transfer 30% to stakers through incentive module
-	// keep remaining 10% on fees treasury
-	// TODO: swap fees to pool FeeDenom (this is normally USDC)
+func (k Keeper) OnCollectFee(ctx sdk.Context, pool types.Pool, fee sdk.Coins) error {
+	poolRevenueAddress := types.NewPoolRevenueAddress(pool.PoolId)
```
Comment (kenta-elys) [inline x/amm/keeper/fee.go:23]: @jelysn I guess we have agreed on defining this function in incentive module side, didn't we?

**Review by cosmic-vagabond — APPROVED** [review]
LGTM, just added one minor comment about params

File: x/amm/keeper/fee.go [hunk 2]
```diff
@@ -2,9 +2,12 @@ package keeper
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/elys-network/elys/x/amm/types"
 )
 
+var WeightRecoveryPortion = sdk.NewDecWithPrec(10, 2) // 10%
```
[the remark this answers is not in the document]
Comment (jelysn) [reply x/amm/keeper/fee.go:9]: I am still not sure to make this 10% to be per pool or module parameter, just put as constant for now.

**Comment by cosmic-vagabond** [comment]
duplicates https://github.com/elys-network/elys/pull/137

**Pull request closed without merging**